### PR TITLE
perl-for-cntrl-chars

### PR DESCRIPTION
**Pull request by joesuf4**

### CODE REVIEW

File: kubectl-node_shell [hunk 1]
```diff
@@ -75,7 +75,7 @@ done
 if [ $# -gt 0 ]; then
   while [ $# -gt 0 ]; do
     cmd="$cmd, \"$(echo "$1" | \
-      awk '{gsub(/\\/,"\\\\");gsub(/"/,"\\\"");gsub(/$/,"\\n");printf last}{last=$0} END{gsub(/\\n/,"",last);printf last}' \
+    perl -pl -0777 -e 's/([\\"])/\\$1/g;s/\n$//;s/\n/\\n/g;s/([\x01-\x1f])/sprintf "\\u%04x",ord $1/ge' \
```
Comment (kvaps) [inline kubectl-node_shell:78]: Unfortunately `perl` is not installed on many systems by default, let's try using as less dependencies as we can.
Awk was a nice solution, and persist almost everywhere, so why not using it?

**Review by kvaps — COMMENTED** [review]
Hi, @joesuf4 sorry for late answer. I'm going to review your PR

File: kubectl-node_shell (hunk 1 above)
Comment (joesuf4) [reply kubectl-node_shell:78]: Because it doesn't handle control-characters like 0x2b, which I need?  https://datatracker.ietf.org/doc/html/rfc7159#page-8

Comment (joesuf4) [reply kubectl-node_shell:78]: Exactly which systems have kubectl installed, but not perl?

**Comment by joesuf4** [comment]
If you want to implement most of the JSON spec faithfully, as it relates to the bulk of 7-bit ascii, you need to use something with more power than awk.

If you just want to support my use case, and stick to awk, that's also fine with me.   This MR has that developed in the final commit.

File: kubectl-node_shell (hunk 1 above)
Comment (kvaps) [reply kubectl-node_shell]: Many of them, for example centos, alpine linux, archlinux have no it out of box.

**Comment by kvaps** [comment]
@joesuf4 thanks! Could you also implement the test case you solving by this change?

**Comment by joesuf4** [comment]
Done.

**Comment by kvaps** [comment]
Perfect, thank you for contribution!

**Pull request merged**